nixos/modules/services/monitoring/beszel-agent.nix

```
@@ -0,0 +1,107 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.beszel.agent;
+in
+{
+  meta.maintainers = with lib.maintainers; [ BonusPlay ];
+
+  options.services.beszel.agent = {
+    enable = lib.mkEnableOption "beszel agent";
+
+    package = lib.mkPackageOption pkgs "beszel" { };
+
+    user = lib.mkOption {
+      default = "beszel-agent";
+      type = lib.types.str;
+      description = "User the beszel agent runs under.";
+    };
+    group = lib.mkOption {
+      default = "beszel-agent";
+      type = lib.types.str;
+      description = "Group the beszel agent runs under.";
+    };
+
+    environment = lib.mkOption {
+      type = lib.types.attrs;
+      default = { };
+      description = ''
+        Environment variables for configuring the beszel-agent service.
+        This field will end up public in /nix/store, for secret values use `environmentFile`.
+      '';
+    };
+    environmentFile = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      description = ''
+        File path containing environment variables for configuring the beszel-agent service in the format of an EnvironmentFile. See {manpage}`systemd.exec(5)`.
+      '';
+    };
+    extraPath = lib.mkOption {
+      type = lib.types.listOf lib.types.package;
+      default = [ ];
+      description = ''
+        Extra packages to add to beszel path (such as nvidia-smi or rocm-smi).
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.beszel-agent = {
+      description = "Beszel Agent";
+
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network-online.target" ];
+      after = [ "network-online.target" ];
+
+      environment = cfg.environment;
+      path = cfg.extraPath;
+
+      serviceConfig = {
+        ExecStart = ''
+          ${lib.getExe cfg.package}/bin/beszel-agent
```

```suggestion
          ${cfg.package}/bin/beszel-agent
```
This won't work, since the the `beszel` package has no `mainProgram` attribute. The same goes for the `hub` config. If the beszel ever gets split into an agent and hub package, then you could use `lib.getExe`.

---

Not to mention I left out `/bin/beszel-agent` at the end. Btw, should I add you as a maintainer to those modules?

---

Sure, I do care enough about the project to help out maintaining it :)